win32/tests/srw_lock_win32_int/srw_lock_win32_int.c

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include <stdlib.h>
+
+#include "windows.h"
+
+#include "macro_utils/macro_utils.h"
+
+#include "testrunnerswitcher.h"
+#include "umock_c/umock_c.h"
+
+#include "c_pal/srw_lock.h"
+
+static TEST_MUTEX_HANDLE test_serialize_mutex;
+
+static const char* TEST_LOCK_NAME = "test_lock";
+
+BEGIN_TEST_SUITE(TEST_SUITE_NAME_FROM_CMAKE)
+
+TEST_SUITE_INITIALIZE(suite_init)
+{
+
+    test_serialize_mutex = TEST_MUTEX_CREATE();
+    ASSERT_IS_NOT_NULL(test_serialize_mutex);
+}
+
+TEST_SUITE_CLEANUP(suite_cleanup)
+{
+    TEST_MUTEX_DESTROY(test_serialize_mutex);
+
+}
+
+TEST_FUNCTION_INITIALIZE(method_init)
+{
+    if (TEST_MUTEX_ACQUIRE(test_serialize_mutex))
+    {
+        ASSERT_FAIL("Could not acquire test serialization mutex.");
+    }
+}
+
+TEST_FUNCTION_CLEANUP(method_cleanup)
+{
+    TEST_MUTEX_RELEASE(test_serialize_mutex);
+}
+
+TEST_FUNCTION(srw_lock_create_succeeds)
+{
+    ///act
+    SRW_LOCK_HANDLE lock_handle = srw_lock_create(false, TEST_LOCK_NAME);
+
```

Comment:
If we have both win32 and linux implemented, I reckon these int tests should be common for both.

Reply:
So they shoul dnot be in the win32 folder, right?